PhpAmqpLib/Wire/AMQPWriter.php

```diff
@@ -302,6 +308,12 @@ public function write_shortstr($s)
      */
     public function write_longstr($s)
     {
+        if ($s === null) {
+            $this->write_long(0);
+
+            return $this;
+        }
+
         $this->write_long(mb_strlen($s, 'ASCII'));
```

Comment:
Failed test: https://github.com/javer/php-amqplib/runs/3210161338?check_suite_focus=true#step:11:30
```
1) PhpAmqpLib\Tests\Unit\Message\AMQPMessageTest::serialize_properties with data set #9 (array(array('')), array(array(array('S', null))))
mb_strlen(): Passing null to parameter #1 ($string) of type string is deprecated

/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPWriter.php:305
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPWriter.php:435
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPWriter.php:365
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPWriter.php:383
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Message/AMQPMessage.php:529
/home/runner/work/php-amqplib/php-amqplib/tests/Unit/Message/AMQPMessageTest.php:21
```

Since PHP 8.1 passing null to non-nullable internal function parameters is deprecated: https://php.watch/versions/8.1/internal-func-non-nullable-null-deprecation

Reply:
I guess a unit test is missing :)

Reply:
@ramunasd I've added unit test for these cases and fixed the logic.